source/server/admin/admin.cc

```diff
@@ -376,22 +376,22 @@ void AdminImpl::writeClustersAsJson(Buffer::Instance& response) {
 
 // TODO(efimki): Add support of text readouts stats.
 void AdminImpl::writeClustersAsText(Buffer::Instance& response) {
-  for (auto& cluster : server_.clusterManager().clusters()) {
-    addOutlierInfo(cluster.second.get().info()->name(), cluster.second.get().outlierDetector(),
-                   response);
-
-    addCircuitSettings(
-        cluster.second.get().info()->name(), "default",
-        cluster.second.get().info()->resourceManager(Upstream::ResourcePriority::Default),
-        response);
-    addCircuitSettings(
-        cluster.second.get().info()->name(), "high",
-        cluster.second.get().info()->resourceManager(Upstream::ResourcePriority::High), response);
-
-    response.add(fmt::format("{}::added_via_api::{}\n", cluster.second.get().info()->name(),
-                             cluster.second.get().info()->addedViaApi()));
-    for (auto& host_set : cluster.second.get().prioritySet().hostSetsPerPriority()) {
+  for (auto& cluster_pair : server_.clusterManager().clusters()) {
```

Comment:
```suggestion
  for (const auto& [cluster_name, cluster] : server_.clusterManager().clusters()) {
```

Reply:
I suppose this is the feature on C++ 17. Does envoy support C++ 17 now?

Reply:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/envoy-dev/lons-wIFeNk/dBNXOUxWAwAJ

Reply:
> https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/envoy-dev/lons-wIFeNk/dBNXOUxWAwAJ

Oh wow, thanks for sharing this!